src/tests/stress/savina/10.ConcurrentSortedLinkedList/Main.enc

```diff
@@ -0,0 +1,171 @@
+-- https://github.com/shamsimam/savina/blob/master/src/main/java/edu/rice/habanero/benchmarks/concsll/SortedLinkedList.java
+-- https://github.com/shamsimam/savina/blob/master/src/main/scala/edu/rice/habanero/benchmarks/concsll/SortedListAkkaActorBenchmark.scala
+-- Sequential sorted linked list ported from the first file, then it's wrapped
+-- inside an actor just like how it's done in the second file
+
+import Random
+
+read trait Ord
+  require def compare(o:Ord) : int
+end
+
+read class Integer : Ord(i)
+  val i:int
+
+  def init(i:int) : unit
+    this.i = i
+  end
+
+  def compare(other:Ord) : int
+    val y = EMBED (Integer) (void*)#{other}; END
```

Comment:
You could get rid of this `EMBED` if you wrote `Ord` using the following pattern:
```
read trait Ord[self]
  require def compare(x : self) : int
end

read class Integer : Ord[Integer](i)
  val i:int

  def init(i:int) : unit
    this.i = i
  end

  def compare(other:Integer) : int
    this.i - other.i
  end
end
```